### PR TITLE
Added filterer to tables

### PR DESCRIPTION
**Pull request by MattPutnam**
Filterer is controlled by `:filterable?` property, defaults to `true`.

Columns filter by string value, or a specified :`filter-by` function taking the column value.  Filtering can be disabled for a column by setting `:filter-by :none`.

Filtering has been added to all tables.  The only changes needed to them was to specify a custom filterer for columns that were not `str`-able.

### CODE REVIEW

File: src/cljs/org/broadinstitute/firecloud_ui/utils.cljs [hunk 1]
```diff
@@ -26,6 +26,10 @@
   ([s what start-index] (.indexOf s what start-index)))
 
 
+(defn contains-ignore-case [s what]
+  (<= 0 (str-index-of (.toLowerCase s) (.toLowerCase what))))
```
Comment (dmohs) [inline src/cljs/org/broadinstitute/firecloud_ui/utils.cljs]: Might be worth using `clojure.string/lower-case`.

**Comment by eddiebroad** [comment]
overall :+1:

File: src/cljs/org/broadinstitute/firecloud_ui/common/table.cljs [hunk 2]
```diff
@@ -170,19 +172,30 @@
...
      (let [paginator-above (= :above (:paginator props))
            paginator-below (= :below (:paginator props))
            paginator [Paginator {:ref "paginator"
-                                 :num-rows (count (:data props))
+                                 :num-rows (count (:filtered-data @state))
                                  :onChange #(react/call :handle-pagination-change this)}]]
        [:div {}
+        (when (:filterable? props)
+          (let [apply-filter #(swap! state assoc :filtered-data
+                               (react/call :filter-data this (trim (.-value (.getDOMNode (@refs "filter-field"))))))]
```
Comment (dmohs) [inline src/cljs/org/broadinstitute/firecloud_ui/common/table.cljs]: `(react/call :filter-data this (-> (@refs "filter-field") .getDOMNode .-value trim))` would be another option if you like.

**Comment by dmohs** [comment]
:+1: Really nice to have the filtering everywhere. Good to merge.

File: src/cljs/org/broadinstitute/firecloud_ui/common/table.cljs (hunk 2 above)
Comment (eddiebroad) [inline src/cljs/org/broadinstitute/firecloud_ui/common/table.cljs]: more clear now

Comment (eddiebroad) [inline src/cljs/org/broadinstitute/firecloud_ui/common/table.cljs]: it's more clear now

**Pull request merged**